app/controllers/evidence_hub_controller.rb

```diff
@@ -2,17 +2,27 @@ class EvidenceHubController < ApplicationController
   DOCUMENT_TYPES = ['Insight'].freeze
 
   def index
-    @search_form = EvidenceSummarySearchForm.new(search_params)
-    documents = Mas::Cms::Document.all(params: parse_params)
+    @search_form = EvidenceSummarySearchForm.new(form_params)
+    documents = Mas::Cms::Document.all(params: search_params)
     @evidence_summaries = EvidenceSummary.map(documents)
   end
 
   private
 
-  def search_params
-    {
+  def form_params
+    @search_params ||= {
```

Comment:
method name `form_param`  - variable name `@search_params` ... ? 😕

Reply:
I'm having a brain dead morning, so I may have missed something obvious and stupid here...
should the variable here not be name `@form_params` rather than `@search_params`? does it need to be a global variable? I can't see it being called anywhere else.
Can/should the setting of the variable to begin with be separated out of this method?

Reply:
Its a little confusing and we changed the variable names so that:

- the parameter required for instantiating the EvidenceSummarySearchForm is called form_params and
- the parameter required for actually searching the documents (Mas::Cms::Document.all) is called search params

We need to cater for several scenarios as detailed in the description and **In all cases, the code should send the document type to the CMS API to only retrieved Evidence Hub documents.** Also the parameters for the form itself may include the reset link.

Reply:
but the parameter for searching documents (Mas::Cms::Document.all) is calling the *method* `search_params` not the *variable* `@search_params`

Reply:
my point is that the method `form_params`
- sets the variable `@search_params`, **and** manipulates that variable. 2 responsibilities?
- returns the variable `@search_params`, surely this would make more sense to be `@form_params`? 
I'm also unsure why it is a global variable?

Reply:
The points were good raised. I renamed the variable to `form_params` and remove as instance variable which makes total sense. I think now the code is concise. 

`form_params` is to build the form object and the `search_params` is to convert to a CMS format.

Reply:
Thanks for the clarification. I think the change makes a lot more sense